.ci/aws-instances-reaper.groovy

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env groovy
+
+@Library('apm@current') _
+
+pipeline {
+  agent { label 'ubuntu-20' }
+  environment {
+    REPO = 'e2e-testing'
+    BASE_DIR = "src/github.com/elastic/${env.REPO}"
+    HOME = "${env.WORKSPACE}"
+    NOTIFY_TO = credentials('notify-to')
+    PIPELINE_LOG_LEVEL = 'INFO'
+    JOB_GIT_CREDENTIALS = "f6c7695a-671e-4f4f-a331-acdce44ff9ba"
+    AWS_PROVISIONER_SECRET = 'secret/observability-team/ci/elastic-observability-aws-account-auth'
+    AWS_EC2_INSTANCES_TAG= 'e2e-testing-vm'
+  }
+  options {
+    timeout(time: 1, unit: 'HOURS')
+    buildDiscarder(logRotator(numToKeepStr: '20', artifactNumToKeepStr: '20'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+    rateLimitBuilds(throttle: [count: 60, durationName: 'hour', userBoost: true])
+    quietPeriod(10)
+  }
+  triggers {
+    cron '0 0 * * 0'
+  }
+  stages {
+    stage('Checkout') {
+      steps {
+        deleteDir()
+        gitCheckout(basedir: "${BASE_DIR}",
+          branch: "main",
+          repo: "https://github.com/elastic/${REPO}.git",
+          credentialsId: "${JOB_GIT_CREDENTIALS}"
+        )
+        stash allowEmpty: true, name: 'source', useDefaultExcludes: false
+      }
+    }
+    stage('Reap AWS instances'){
+      environment {
+        HOME = "${env.WORKSPACE}/${BASE_DIR}"
+      }
+      steps {
+        deleteDir()
+        unstash 'source'
+        withAWSEnv(secret: "${env.AWS_PROVISIONER_SECRET}") {
```

Comment:
@v1v should we force installation?
```suggestion
        withAWSEnv(secret: "${env.AWS_PROVISIONER_SECRET}", forceInstallation: true) {
```

Reply:
That's a good question. IIRC, I saw some workers in the past that their python was not correctly installed and somehow the aws cli failed to be used, if this pipeline can install a fresh aws cli installation this should avoid any environmental issues in the workers.

Short answer, yes if the  speed is not important

Reply:
Missing the `dir(base_dir)` closure

Reply:
Do we still need it? We are simply running aws cli without any info from the repo

Reply:
> Short answer, yes if the speed is not important

Yes, speed is not important :)

Reply:
That's correct. Will update the PR now

Reply:
Added in https://github.com/elastic/e2e-testing/pull/2001/commits/600eb0a0f6f7d1845dc684087c3d5cb34b495351